File: test/language/packages/python/PackageNameConflictTest.py
import unittest

import zserio

from testutils import getZserioApi

class PackageNameConflictTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "package_name_conflict.zs")

    def testPackageNameConflictLocal(self):
        # just test that PackageNameConflictLocal uses correct Blob
        packageNameConflictLocal = self.api.PackageNameConflictLocal(self.api.Blob(13))
        bitBuffer = zserio.serialize(packageNameConflictLocal)
        readPackageNameConflictLocal = zserio.deserialize(self.api.PackageNameConflictLocal, bitBuffer)

        self.assertEqual(13, packageNameConflictLocal.blob.value)
        self.assertEqual(packageNameConflictLocal.blob.value, readPackageNameConflictLocal.blob.value)

    def testPackageNameConflictImported(self):
        # just test that PackageNameConflictImported uses correct Blob
        packageNameConflictImported = self.api.PackageNameConflictImported(
            self.api.package_name_conflict.Blob("test")
        )
        bitBuffer = zserio.serialize(packageNameConflictImported)
        readPackageNameConflictImported = zserio.deserialize(self.api.PackageNameConflictImported, bitBuffer)

        self.assertEqual("test", packageNameConflictImported.blob.value)
        self.assertEqual(packageNameConflictImported.blob.value,
                         readPackageNameConflictImported.blob.value)

File: test/language/packages/cpp/PackageNameConflictTest.cpp
#include "gtest/gtest.h"
#include "package_name_conflict/PackageNameConflictImported.h"
#include "package_name_conflict/PackageNameConflictLocal.h"

namespace package_name_conflict
{

TEST(PackageNameConflictTest, packageNameConflictLocal)
{
    // just test that PackageNameConflictLocal includes correct Blob
    PackageNameConflictLocal packageNameConflictLocal{Blob{13}};

    zserio::BitBuffer bitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    packageNameConflictLocal.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    PackageNameConflictLocal readPackageNameConflictLocal(reader);

    ASSERT_EQ(13, packageNameConflictLocal.getBlob().getValue());
    ASSERT_EQ(packageNameConflictLocal.getBlob().getValue(), readPackageNameConflictLocal.getBlob().getValue());
}

TEST(PackageNameConflictTest, packageNameConflictImported)
{
    // just test that PackageNameConflictImported includes correct Blob
    PackageNameConflictImported packageNameConflictImported{
            ::package_name_conflict::package_name_conflict::Blob{"test"}};

    zserio::BitBuffer bitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    packageNameConflictImported.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    PackageNameConflictImported readPackageNameConflictImported(reader);

    ASSERT_EQ("test", packageNameConflictImported.getBlob().getValue());
    ASSERT_EQ(packageNameConflictImported.getBlob().getValue(),
            readPackageNameConflictImported.getBlob().getValue());
}

} // namespace package_name_conflict
